src/racket/synthesis.rkt

```diff
@@ -469,8 +473,10 @@
     (define input-node (get-node graph input-id))
     (append (for/list ([delta-name (send input-node get-delta-names)])
       (define intermediate-ids '())
-      (for ([output-id (get-ids graph)] #:when (not (equal? output-id input-id)))
+      (for ([output-id (get-ids graph)] #:when (path? graph output-id input-id))
```

Comment:
Currently we are running the same BFS multiple times -- a more efficient algorithm would compute the BFS once, and then iterate over it. This doesn't matter since our graphs are tiny, but add a comment noting the inefficiency.